Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/certificate/Dcc.kt

```diff
@@ -1,9 +1,14 @@
 package de.rki.coronawarnapp.covidcertificate.common.certificate
 
 import com.google.gson.annotations.SerializedName
+import org.joda.time.DateTime
 import org.joda.time.LocalDate
+import org.joda.time.format.DateTimeFormat
+import org.joda.time.format.DateTimeFormatterBuilder
+import org.joda.time.format.ISODateTimeFormat
+import timber.log.Timber
 
-interface Dcc<PayloadType : Dcc.Payload> {
+abstract class Dcc<PayloadType : Dcc.Payload> {
```

Comment:
I'm just curious: What was wrong with being an interface?

Reply:
Interface doesn't allow backing fields, by making it an abstract class I can move the caching for `dateOfBirth` parsing into the `Dcc` class and reuse it for all types of certificates.